plugins/integration/providers/cielo24/Cielo24ClientHelper.php

```diff
@@ -0,0 +1,164 @@
+		$baseUrl = $baseUrl . "account/login?v=$version";
+		$loginParams = array("username" => $username, "password" => $password);	
+		$loginUrl = $this->addUrlParams($baseUrl, $loginParams);
+	
+		$loginResult = self::sendAPICall($loginUrl);
+		$apiToken = $loginResult->ApiToken;
+			
+		$this->baseEndpointUrl = str_replace("account/login", self::API_CALL_NAME_PATTERN, $baseUrl);
+		$this->baseEndpointUrl .= "&api_token=$apiToken";   
+		
+		$this->supportedLanguages = $cielo24ParamsMap['languages'];
+	}
+	
+	public function getRemoteFinishedJobId($entryId)
+	{
+		$remoteJobsListAPIUrl = str_replace(self::API_CALL_NAME_PATTERN, "job/list", $this->baseEndpointUrl);
+		$listParams = array("ExternalID" => $entryId, "JobStatus" => "Complete");
+		$remoteJobsListAPIUrl = $this->addUrlParams($remoteJobsListAPIUrl, $listParams);
+	
+		$exitingJobsResult = $this->sendAPICall($remoteJobsListAPIUrl);
+		if($exitingJobsResult && isset($exitingJobsResult->ActiveJobs) && count($exitingJobsResult->ActiveJobs))
+		{
+			return $exitingJobsResult->ActiveJobs[0]->JobId;
+		}
+		return false;
+	}
+	
+	public function uploadMedia($flavorUrl, $entryId, $callBackUrl, $spokenLanguage, $priority, $fidelity)
+	{
+		$languageExternalServiceParam = $this->supportedLanguages[$spokenLanguage];
+		
+		//adding a job
+		$createJobAPIUrl = str_replace(self::API_CALL_NAME_PATTERN, "job/new", $this->baseEndpointUrl);
+		$jobCreationParams = array("language" => $languageExternalServiceParam,
+						"external_id" => $entryId
+						);
+		$createJobAPIUrl = $this->addUrlParams($createJobAPIUrl, $jobCreationParams);
+		$jobAdditionResult = $this->sendAPICall($createJobAPIUrl);
+		if($jobAdditionResult && isset($jobAdditionResult->JobId))
+			$jobId = $jobAdditionResult->JobId;
+		else
+			return false;
+		
+		// attaching media to the job
+		$addMediaAPIUrl = str_replace(self::API_CALL_NAME_PATTERN, "job/add_media", $this->baseEndpointUrl);
+		$addMediaParams = array("job_id" => $jobId, "media_url" => $flavorUrl);
+		$addMediaAPIUrl = $this->addUrlParams($addMediaAPIUrl, $addMediaParams);
+		$mediaAdditionResult = $this->sendAPICall($addMediaAPIUrl);
+		if(!$mediaAdditionResult || !isset($mediaAdditionResult->TaskId))
+			return false;
+		
+		//request transcription
+		$requestTransctiptAPIUrl = str_replace(self::API_CALL_NAME_PATTERN, "job/perform_transcription", $this->baseEndpointUrl);
+		$requestParams = array("job_id" => $jobId,
+					"transcription_fidelity" => $fidelity,
+					"priority" => $priority,
+					"callback_url" => $callBackUrl
+					);
+		$requestTransctiptAPIUrl = $this->addUrlParams($requestTransctiptAPIUrl, $requestParams);
+		$requestTranscriptionResult = $this->sendAPICall($requestTransctiptAPIUrl);
+		if(!$requestTranscriptionResult || !isset($requestTranscriptionResult->TaskId))
+			return false;
+			
+		return true;
+	}
+	
+	private function sendAPICall($url, $noDecoding = false)
+	{
+		KalturaLog::debug("sending API call - $url");
```

Comment:
@shpiNi you should pass the API name and the parameters to this function and build the full URL internally.
This will solve the `API_CALL_NAME_PATTERN` constant.